jwst/datamodels/dqflags.py

```diff
@@ -1,41 +1,108 @@
-pixel = {'GOOD': 0,
-         'DO_NOT_USE': 1,
-         'SATURATED': 2,
-         'JUMP_DET': 4,
-         'DROPOUT': 8,
-         'RESERVED_1': 16,
-         'RESERVED_2': 32,
-         'RESERVED_3': 64,
-         'RESERVED_4': 128,
-         'UNRELIABLE_ERROR': 256,
-         'NON_SCIENCE': 512,
-         'DEAD': 1024,
-         'HOT': 2048,
-         'WARM': 4096,
-         'LOW_QE': 8192,
-         'RC': 16384,
-         'TELEGRAPH': 32768,
-         'NONLINEAR': 65536,
-         'BAD_REF_PIXEL': 131072,
-         'NO_FLAT_FIELD': 262144,
-         'NO_GAIN_VALUE': 524288,
-         'NO_LIN_CORR': 1048576,
-         'NO_SAT_CHECK': 2097152,
-         'UNRELIABLE_BIAS': 4194304,
-         'UNRELIABLE_DARK': 8388608,
-         'UNRELIABLE_SLOPE': 16777216,
-         'UNRELIABLE_FLAT': 33554432,
-         'OPEN': 67108864,
-         'ADJ_OPEN': 134217728,
-         'UNRELIABLE_RESET': 268435456,
-         'MSA_FAILED_OPEN': 536870912,
-         'OTHER_BAD_PIXEL': 1073741824,
-         'REFERENCE_PIXEL': 2147483648
+""" JWST Data Quality Flags
+
+The definitions are documented in the JWST RTD:
+
+https://jwst-pipeline.readthedocs.io/en/latest/jwst/references_general/references_general.html#data-quality-flags
+
+
+Implementation
+-------------
+
+The flags are implemented as "bit flags": Each flag is assigned a bit position
+in a byte, or multi-byte word, of memory. If that bit is set, the flag assigned
+to that bit is interpreted as being set or active.
+
+The data structure that stores bit flags is just the standard Python `int`,
+which provides 32 bits. Bits of an integer are most easily referred to using
+the formula `2**bit_number` where `bit_number` is the 0-index bit of interest.
+"""
+from astropy.nddata.bitmask import interpret_bit_flags as ap_interpret_bit_flags
+
+from jwst.lib.basic_utils import multiple_replace
+
+# Pixel-specific flags
+pixel = {'GOOD':             0,      # No bits set, all is good
+         'DO_NOT_USE':       2**0,   # Bad pixel. Do not use.
+         'SATURATED':        2**1,   # Pixel saturated during exposure
+         'JUMP_DET':         2**2,   # Jump detected during exposure
+         'DROPOUT':          2**3,   # Data lost in transmission
+         'RESERVED_1':       2**4,   #
+         'RESERVED_2':       2**5,   #
+         'RESERVED_3':       2**6,   #
+         'RESERVED_4':       2**7,   #
```

Comment:
I suggest that for now we utilize one of the reserved bits and call it "OUTLIER" and have the `outlier_detection` step set that informational flag, as well as "DO_NOT_USE", for the outliers that it detects. That gives us a unique flag from "JUMP_DET".

Reply:
Sounds like a plan. The first reserved bit will be used.

Just for FYI, as implied earlier when determined what flags were being set, the DO_NOT_USE flag was never set for the marked outliers. This will be fixed also.

So, all discovered outliers will have DO_NOT_USE + OUTLIER set in the DQ array.